dev_tools/interactive/refinement_app/viewmodel.py

```diff
@@ -0,0 +1,397 @@
+from shiny.express import ui
+from shiny import reactive
+import pandas as pd
+import gxhistory
+import os
+from gsasIImodel import (
+    hist_export,
+    gsas_load_gpx,
+    load_phase_constraints,
+    load_histogram_parameters,
+)
+import matplotlib.pyplot as plt
+
+"""contains all reactive events functions and variables
+and processes all logic from the ui, GSASII and galaxy history models.
+note functions defined here cannot directly access inputs from view.
+these have to be passed as inputs to the function."""
+
+# reactive value parameters for anything that needs to be passed on
+# as a side effect from reactive functions
+gpx = reactive.value()
+instreflist = reactive.value()
+instparams = reactive.value()
+sampreflist = reactive.value()
+sampleparams = reactive.value()
+inputgpxfile = reactive.value()
+
+x = reactive.value()
+y = reactive.value()
+ycalc = reactive.value()
+dy = reactive.value()
+bkg = reactive.value()
+
+histdata = reactive.value()
+constraints = reactive.value()
+
+# initial values for sidebar selections when no project is loaded
+# reactive values for when projects are loaded
+gpx_choices = {"init":
+               "update the history before loading a new project"}
+select_gpx_choices = reactive.value(gpx_choices)
+
+phase_choices = {"init":
+                 "Load a project before selecting a phase"}
+select_phase_choices = reactive.value(phase_choices)
+
+hist_choices = {"init":
+                "Load a project before selecting a histogram"}
+select_hist_choices = reactive.value(hist_choices)
+view_hist_choices = {"init":
+                     "Load a project before selecting a histogram"}
+select_view_hist = reactive.value(view_hist_choices)
+
+view_proj_choices = {"Notebook": "Notebook", "Controls": "Controls",
+                     "Constraints": "Constraints", "Restraints": "Restraints",
+                     "Rigid Bodies": "Rigid Bodies"}
+
+
+def updatenav(tab):
```

Comment:
## Formatting/Style
Most places are using `snake_case` but some of the functions here aren't cased and so are a bit hard to read. Not going to comment them all, but it seems to just be this file.

```suggestion
def update_nav(tab):
```